File: packages/ink-swift/Sources/InkCalligrapher/Resources.swift
import Foundation

/// The weights committed with the repo (packages/ink-calligrapher/assets),
/// bundled into this target as a resource — consumers who add the package
/// from git load the model with no downloads or file wrangling:
///
///     let model = try CalligrapherModel(assets: parseCalligrapherWeights(bundledCalligrapherWeights()))
public func bundledCalligrapherWeights() throws -> Data {
    guard let url = Bundle.module.url(forResource: "calligrapher-v1", withExtension: "bin") else {
        throw CalligrapherError.missingBundledWeights
    }
    return try Data(contentsOf: url)
}

File: packages/ink-kotlin/ink-calligrapher/src/main/kotlin/com/trylonghand/ink/calligrapher/Resources.kt
package com.trylonghand.ink.calligrapher

/**
 * The weights committed with the repo (packages/ink-calligrapher/assets),
 * bundled into this module's JAR as a resource — consumers who add the
 * package load the model with no downloads or file wrangling:
 *
 *     val model = CalligrapherModel(parseCalligrapherWeights(bundledCalligrapherWeights()))
 */
public fun bundledCalligrapherWeights(): ByteArray {
    val stream = object {}.javaClass.getResourceAsStream("/calligrapher-v1.bin")
        ?: throw CalligrapherError.MissingBundledWeights()
    return stream.use { it.readBytes() }
}
